### PR TITLE
Add Support for Different Marketplace Tags in Detailed Description and Readme Files

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/46588

## Description
Add handling for `<~XSOAR></~XSOAR>` and `<~XSIAM>></~XSIAM>` tags.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 * enhanced ***validate*** command to list all command names affected by a backward compatibility break, instead of only one.
 * Added support for Wizard content item in the **format**, **validate**, **upload**, **create-id-set**, **find-dependecies** and **create-content-artifacts** commands.
 * Added a new flag to the **validate** command, allowing to run specific validations.
+* Added support in **unify** and **create-content-artifacts** for slightly different documentations (detailed description + readme) depending on the marketplace version.
```
Comment (dorschw) [inline CHANGELOG.md:11]:
```suggestion
* Added support in **unify** and **create-content-artifacts** for showing content item documentations (detailed description + readme) depending on the marketplace version.
```

File: demisto_sdk/commands/common/content/objects/pack_objects/readme/readme.py [hunk 2]
```diff
@@ -27,6 +28,17 @@ def mention_contributors_in_readme(self):
...
 
+    def handle_marketplace_tags(self):
+        try:
+            with open(self._path, 'r+') as f:
+                text = f.read()
+                f.seek(0)
+                f.write(MARKETPLACE_TAG_PARSER.parse_text(text))
+                f.truncate()
+        except Exception as e:
+            print(e)
```
Comment (dorschw) [inline demisto_sdk/commands/common/content/objects/pack_objects/readme/readme.py:39]: don't we want to raise it? or at least print_error?

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -71,6 +71,86 @@ class LOG_COLORS:
...
+        return self._marketplace
+
+    @marketplace.setter
+    def marketplace(self, marketplace):
+        self._marketplace = marketplace
+        self._should_remove_xsoar_text = marketplace != MarketplaceVersions.XSOAR.value
+        self._should_remove_xsiam_text = marketplace != MarketplaceVersions.MarketplaceV2.value
+
+    def parse_text(self, text):
+        return self._xsiam_inline_parser.parse(
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:137]: speaking of efficiency, adding a short explanation here (of why this order is used) may be helpful for future devs

**Review by dorschw — APPROVED** [review]
nice

File: demisto_sdk/commands/common/tools.py (hunk 3 above)
Comment (DeanArbel) [reply demisto_sdk/commands/common/tools.py:137]: True

File: demisto_sdk/commands/common/content/objects/pack_objects/readme/readme.py (hunk 2 above)
Comment (DeanArbel) [reply demisto_sdk/commands/common/content/objects/pack_objects/readme/readme.py:39]: Was lazy and copied what the above method does. I think in this case a normal print should be ok, but it's better to log it properly.

**Pull request merged**